api/src/main/java/com/datastrato/gravitino/exceptions/NotPartitionedTableException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.exceptions;
+
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+
+/** Exception thrown when a partitioned table does not exist. */
+public class NotPartitionedTableException extends NotFoundException {
```

Comment:
I don't think `NotPartitionedTableException` is a subclass of `NotFoundException`, from my perspective, `NotPartitionedTableException` will be thrown in the case a table exists but it's NOT a partitioned table.

Reply:
It makes sense to me, I'll fix it later.